### PR TITLE
Blank publish allow

### PR DESCRIPTION
**Pull request by TakenPilot**
Dependent on this PR: https://github.com/nymag/amphora/pull/207

New Feature:
- When PUTing to `@published`, if no data is provided, will use `@latest` for pages and components

### CODE REVIEW

File: test/mocha.opts [hunk 1]
```diff
@@ -1,4 +1,4 @@
---reporter dot
+--reporter spec
```
Comment (cruzanmo) [inline test/mocha.opts]: Interesting -- prettier?

**Comment by cruzanmo** [comment]
Looks good. What is a use case for this feature?

File: lib/services/components.js [hunk 2]
```diff
@@ -268,6 +268,23 @@ function cascadingPut(uri, data, locals) {
 }
 
 /**
+ *
+ * @param uri
+ * @param data
+ * @param locals
+ * @returns {*}
+ */
```
Comment (cruzanmo) [inline lib/services/components.js]: Could add types to this JSDoc

File: test/mocha.opts (hunk 1 above)
Comment (TakenPilot) [inline test/mocha.opts]: Nah, I'll change this back.

File: lib/services/components.js (hunk 2 above)
Comment (TakenPilot) [inline lib/services/components.js]: Will add.

**Comment by TakenPilot** [comment]
Ask Nelson about the use-case he needs.  But mostly just a shortcut, since passing empty data to a component is _always_ wrong, we're going to pretend that it has a use by assuming they meant `@latest`.  It'll allow us to write easier scripts to publish content without having to fetch `@latest` first.

File: test/mocha.opts (hunk 1 above)
Comment (yoshokatana) [inline test/mocha.opts]: b-but I like spec!

**Comment by yoshokatana** [comment]
Yup, it'll allow us to do re-publishing of content a lot easier, and allow us to easily publish components. The most pressing use-case for this is of course publishing layouts, but it'll probably be used for more things.

File: lib/services/components.js [hunk 3]
```diff
@@ -268,6 +268,23 @@ function cascadingPut(uri, data, locals) {
...
+function publish(uri, data, locals) {
+  if (data && _.size(data) > 0) {
+    return cascadingPut(uri, data, locals);
```
Comment (yoshokatana) [inline lib/services/components.js]: niiiiiice

**Comment by yoshokatana** [comment]
:+1:

**Comment by yoshokatana** [comment]
even better! :+1: :+1: :shipit:

**Comment by cruzanmo** [comment]
+1

**Pull request merged**